dev_requirements.txt

```diff
@@ -18,3 +20,16 @@ myst-parser~=0.13.3
 pybtex~=0.22.2
 pydata-sphinx-theme~=0.3.1
 notebook~=6.1.5
+
+# Packages needed for individual qiskit install
```

Comment:
Is this to say: If we only install terra, aer, and ibmq-provider, these are the packages that are missing compared to when we install qiskit? Is it a minimal set? Are they all necessary?

Reply:
Yes. I compared the packages installed after a full install to the ones installed individually. Since individual install lead to some tests failing, I just added the packages listed here. I did not test all of them individually to check absence of which ones led to errors.

Reply:
Thanks @purva-thakre. Can you try this in your environment to find a minimal set?

```bash
pip install pipdeptree
pipdeptree --warn silence | grep -E '^\w+'
```

https://pypi.org/project/pipdeptree/

Reply:
@rmlarose Thanks for pointing me towards `pipdeptree`. It seems I did not need the packages that were listed previously. The tests were failing because `terra` was not installed before `aer`. There's an installation order and `pip dependency resolver` has issues with switching to a higher version of dependency for `terra` when a lower one was installed for `aer`. 

![image](https://user-images.githubusercontent.com/66048318/114227313-74b36980-993a-11eb-81b6-1d689ae7c396.png)

For this reason, minimal `qiskit` install by specifying `aer` only might not be a good idea. I have specified terra, aer and ibmq-provider exactly.

Reply:
This is good to know, thanks!